### PR TITLE
Small fix of `fusion.pyx` and docs

### PR DESCRIPTION
**Pull request by xuzijian629**
For docs, I replaced all `cupy.~` by `cp.~` in `kernel.rst` when it is used in code example, as `basics.rst` does so.

### CODE REVIEW

File: cupy/core/fusion.pyx [hunk 1]
```diff
@@ -533,7 +533,7 @@ class _FusionHistory(object):
                       (float, bool, complex, numpy.generic)):
             var = self._fresh_local(numpy.dtype(type(arg)), const_value=arg)
             return _FusionVarScalar(var, -1, self._has_reduction())
-        raise Exception('Unsupported type {}'.format(type(type)))
+        raise Exception('Unsupported type {}'.format(type(arg)))
```
Comment (asi1024) [inline cupy/core/fusion.pyx]:
```suggestion
        raise TypeError('Unsupported type {}'.format(type(arg)))
```

File: cupy/core/fusion.pyx [hunk 2]
```diff
@@ -162,7 +162,7 @@ class _FusionOp(object):
 
     def __repr__(self):
         return '<_FusionOp #{}, {} types=[{}]>'.format(
-            self.index, self.submodule.name, ', '.join(self.dtypes))
+            self.index, self.submodule.name, ', '.join(map(str, self.dtypes)))
```
Comment (asi1024) [inline cupy/core/fusion.pyx]: Is this fix correct?

Comment (xuzijian629) [reply cupy/core/fusion.pyx]: The original code raises `TypeError: sequence item 0: expected str instance, numpy.dtype found`.

**Comment by asi1024** [comment]
LGTM. pfnCI, test this please.

**Comment by asi1024** [comment]
Jenkins failures seem unrelated to this PR.

**Comment by asi1024** [comment]
pfnCI, test this please.

**Pull request merged**